File: lib/handler/no_pipelining.rb
# https://github.com/ngocdaothanh/xitrum/commit/31c727f9c125239ef3e25cf1edf2eb3ae24b6515
class Katamari::Handler::NoPipelining < Katamari::Handler::Upstream
  def messageReceived(ctx, e)
    ctx.channel.set_readable(false)
    ctx.send_upstream(e)
  end
end

File: src/main/java/io/katamari/handler/NoPipelining.java
package io.katamari.handler;

import org.jboss.netty.channel.SimpleChannelUpstreamHandler;
import org.jboss.netty.channel.ChannelHandlerContext;
import org.jboss.netty.channel.MessageEvent;
import org.jboss.netty.channel.ExceptionEvent;

public class NoPipelining extends SimpleChannelUpstreamHandler {
  @Override
  public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) throws Exception {
    ctx.getChannel().setReadable(false);
    ctx.sendUpstream(e);
  }

  @Override
  public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws Exception {
    e.getCause().printStackTrace();
    e.getChannel().close();
  }
}